### PR TITLE
Add the ability to set a custom redirect URI.

### PR DESCRIPTION
**Pull request by RicoYao**
[BOXCLient setClientID:@"client-id" clientSecret:@"client-secret" redirectURIString:@"redirectURIString"];

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @RicoYao has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m [hunk 1]
```diff
@@ -34,12 +34,12 @@ @implementation BOXOAuth2Session
 
 #pragma mark - Initialization
 
-
 - (instancetype)initWithClientID:(NSString *)ID secret:(NSString *)secret APIBaseURL:(NSString *)baseURL queueManager:(BOXAPIQueueManager *)queueManager
 {
     if ([self initWithAPIBaseURL:baseURL queueManager:queueManager]) {
         _clientID = ID;
         _clientSecret = secret;
+        _redirectURIString = [NSString stringWithFormat:@"boxsdk-%@://boxsdkoauth2redirect", self.clientID];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m]: Should use the client ID ivar instead of the property getter in init.

**Comment by boxtcarpel** [comment]
+2

:ice_cream:

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m (hunk 1 above)
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m]: Oops, copy-paste-soup. I'll fix.

**Pull request merged**